onnxruntime/core/graph/model.h

```diff
@@ -187,25 +187,36 @@ class Model {
   // Get model's serialization proto data.
   // Save initializer larger than the given threshold (in bytes) into an external binary file
   // with the given name. This function is useful to avoid hitting the size limit of protobuf files.
+  // initializer offset could be page aligned and allocation granularity aligned for mmap support.
   ONNX_NAMESPACE::ModelProto ToGraphProtoWithExternalInitializers(const std::filesystem::path& external_file_name,
                                                                   const std::filesystem::path& file_path,
-                                                                  size_t initializer_size_threshold) const;
+                                                                  size_t initializer_size_threshold,
+                                                                  bool align_offset = false,
+                                                                  size_t align_threshold = 1048576,
+                                                                  size_t allocation_granularity = 65536) const;
```

Comment:
Can we use a struct like AlignmentInfo and keep the defaults in the struct? This way the defaults don't have to be repeated.

Reply:
that's great idea, will do